yogstation/code/game/objects/effects/landmarks.dm

```diff
@@ -62,11 +62,22 @@
 	qdel(src)
 	return TRUE
 
+// Proc to allow you to add coniditions for choosing templates, instead of just randomly picking from the template list.
+// Examples where this would be useful, would be choosing certain templates depending on conditions such as holidays, 
+// Or co-dependent templates, such as having a template for the core and one for the satelite, and swapping AI and comms.git 
+/obj/effect/landmark/stationroom/proc/choose()
+	return safepick(template_names)
+
 /obj/effect/landmark/stationroom/box/bar
 	template_names = list("Bar Trek", "Bar Spacious", "Bar Box", "Bar Casino", "Bar Conveyor", "Bar Diner", "Bar Disco", "Bar Purple")
 	icon = 'yogstation/icons/rooms/box/bar.dmi'
 	icon_state = "bar_box"
 
+/obj/effect/landmark/stationroom/box/bar/choose()
+	if(SSevents.holidays["St. Patrick's Day"])
+		return "Bar Irish"
+	return ..()
```

Comment:
Replace this with . = ..() at the top, you never know when something fucks up and the proc somehow runtimes and you end up without a bar